File: treinando/poo.py

class Vendendor ():
    def __init__ (self, nome):
        self.nome = nome
        self.vendas = 0

    def vendeu (self, vendas):
        self.vendas = vendas
    
    def bat_meta(self, meta ):
        if self.vendas >= meta:
            print (self.nome, 'a meta foi batida!')
        else:
            print (self.nome, 'A meta nao foi batida')

vendedor1 = Vendendor('Ikaro')
vendedor1.vendeu(1000)
vendedor1.bat_meta(500)

File: codigos-go/poo.go
package main

import "fmt"

type Entregador struct {
	nome string
	entregas int
	meta int
}

var entregador Entregador = Entregador{
	nome: "Ikaro",
	entregas: 5,
	meta: 4,
}


func (f Entregador) meta_entrega() {
	if f.entregas >= f.meta {
		fmt.Println(f.nome, "Bateu a meta!")
	} else {
		fmt.Println(f.nome, "Nao bateu a meta!")
	}
}

func main() {
	entregador.meta_entrega()
}